src/SExpr.purs

```diff
@@ -142,12 +151,12 @@ instance Desugarable Expr E.Expr where
 
 instance Desugarable ListRest E.Expr where
    desug :: forall a m. MonadError Error m => BoundedLattice a => ListRest a -> m (E.Expr a)
-   desug (End α) = pure (enil α)
-   desug (Next α s l) = econs α <$> desug s <*> desug l
+   desug (End α) = pure (enil α Nothing)
+   desug (Next α s l) = econs α Nothing <$> desug s <*> desug l
 
    desugBwd :: forall a. BoundedJoinSemilattice a => E.Expr a -> Raw ListRest -> ListRest a
-   desugBwd (E.Constr α _ _) (End _) = End α
-   desugBwd (E.Constr α _ (e1 : e2 : Nil)) (Next _ s l) =
+   desugBwd (E.Constr α _ _ _) (End _) = End α
```

Comment:
I think we could probably enforced the `Nothing` on the incoming `Expr` here because the `absurd` case will handle the (impossible) `Just` case.